metrics.py

```diff
@@ -13,30 +13,12 @@ def macro_f1(y_true, y_pred):
     return f1
 
 
-def precision_at_k(y_true, y_pred_vals, k):
-    # num true labels in top k predictions / k
-    sortd = np.argsort(y_pred_vals)[:,::-1]
-    topk = sortd[:,:k]
-
-    # get precision at k for each example
-    vals = []
-    for i, tk in enumerate(topk):
-        num_true_in_top_k = y_true[i,tk].sum()
-        vals.append(num_true_in_top_k / k)
-
-    return np.mean(vals)
-
-
-def recall_at_k(y_true, y_pred_vals, k):
-    # num true labels in top k predictions / num true labels
-    sortd = np.argsort(y_pred_vals)[:,::-1]
-    topk = sortd[:,:k]
-
-    # get recall at k for each example
-    vals = []
-    for i, tk in enumerate(topk):
-        num_true_in_top_k = y_true[i,tk].sum()
-        denom = y_true[i,:].sum() + 1e-10
-        vals.append(num_true_in_top_k / float(denom))
-
-    return np.mean(vals)
+def precision_recall_at_ks(y_true, y_pred_vals, top_ks):
+    rank_mat = np.argsort(-y_pred_vals)
+    denom = y_true.sum(axis=1) + 1e-10
+    scores = {}
+    for k in top_ks:
+        y_pred = np.take_along_axis(y_true, rank_mat[:,:k], axis=1)
+        scores[f'P@{k}'] = np.mean(np.sum(y_pred, axis=1) / k).item()  # precision at k
+        scores[f'R@{k}'] = np.mean(y_pred.sum(axis=1) / denom).item()  # recall at k
```

Comment:
Improve readability.
```suggestion
    y_pred_rank = np.argsort(-y_pred_vals)
    n_pos = y_true.sum(axis=1)
    scores = {}
    for k in top_ks:
        n_pos_in_top_k = np.take_along_axis(y_true, y_pred_rank[:,:k], axis=1).sum(axis=1)
        scores[f'P@{k}'] = np.mean(n_pos_in_top_k / k).item()  # precision at k
        scores[f'R@{k}'] = np.mean(n_pos_in_top_k / (n_pos + 1e-10)).item()  # recall at k
```

Reply:
y_pred_rank isn't very clear as it's not really ranks. Maybe y_pred_ranked_dataID.
A bit long but it's used only in line 21

Reply:
y_pred_ranked_idx?